apps/billing/tests/test_process_transaction.py

```diff
@@ -69,6 +70,38 @@ def test_create_transaction(self, mock):
 
             self.assertEqual(value, transaction_data[key])
 
+    @mock.patch("requests.post", side_effect=processor_success_response)
+    def test_doest_not_send_transaction_with_zero_as_total_amount(self, mock):
+        """
+        Test that a transaction will not be sent to the processor if the total amound sold is zero.
+        """
+        payload = deepcopy(self.payload)
+        payload["total_amount_exclude_vat"] = 0
+        payload["total_amount_include_vat"] = 0
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token.key)
+
+        log.info(payload)
+        response = self.client.post(self.endpoint, payload, format="json")
+        log.info(response.content)
+        self.assertEqual(response.status_code, 201)
+
+        transaction = Transaction.objects.get(transaction_id=payload["transaction_id"])
+        transaction_data = TransactionSerializer(transaction).data
+        transaction_data["items"] = [
+            TransactionItemSerializer(item).data for item in transaction.transaction_items.all()
+        ]
+
+        for key, value in dict(response.data["data"]).items():
+            if key == "items":
+                for item in value:
+                    same_item = [i for i in transaction_data["items"] if i == item][0]
+                    [self.assertEqual(v, same_item[k]) for k, v in dict(item).items()]
+
+            self.assertEqual(value, transaction_data[key])
+
+        exist_sageX3_transaction: bool = SageX3TransactionInformation.objects.filter(transaction=transaction).exists()
+        self.assertFalse(exist_sageX3_transaction)
```

Comment:
@Tiago-Salles if you are adding this line here, then you need to on the ` test_create_transaction` test, to check that the  SageX3TransactionInformation object has been created.
If not, I have feeling that this could revert very easily, the tests of absent is very tricky. 

On the code, why not just:
```python
 self.assertFalse(SageX3TransactionInformation.objects.filter(transaction=transaction).exists())
```

Reply:
@igobranco Considering that each test contains its own business logic validation, and the fact that receiving a transaction with no earnings it needs to be handled by registering on the database but not sending to the processor, it seems correct to validate that if sending the post request it will act as expected. Could sounds redundant, but a new payload is created with `deepcopy` to be changed according to the needs, after that a new post request is made, that is, it is a different situation.

Reply:
No specific reasons, the change sounds fine.